app/org/sagebionetworks/bridge/play/controllers/ReportController.java

```diff
@@ -214,6 +215,20 @@ public Result saveStudyReport(String identifier) throws Exception {
         
         return createdResult("Report data saved.");
     }
+
+    /**
+     * A similar method as above but specifying study id only for WORKER
+     */
+    public Result saveStudyReportForSpecifiedStudy(String studyId, String identifier) throws Exception {
+        UserSession session = getAuthenticatedSession(WORKER);
+
+        ReportData reportData = parseJson(request(), ReportData.class);
+        reportData.setKey(null); // set in service, but just so no future use depends on it
+
+        reportService.saveStudyReport(new StudyIdentifierImpl(studyId), identifier, reportData);
```

Comment:
You should verify that the studyId is a valid study. Easiest way to do this is to call the study service, you can use the study as the StudyIdentifier key.